tests/unit/storage_rocks.cpp

```diff
@@ -9,111 +9,109 @@
-// auto *acc_ptr = static_cast<memgraph::storage::DiskStorage::DiskAccessor *>(acc.get());
-// const char *value = "garbage";
-// auto vertex = acc_ptr->DeserializeVertex(serialized_vertex, value);
-// ASSERT_EQ(vertex->Gid().AsInt(), 1);
-// ASSERT_EQ(*vertex->HasLabel(memgraph::storage::LabelId::FromUint(1), memgraph::storage::View::OLD), true);
-// }
+TEST_F(RocksDBStorageTest, SerializePropertiesLocalBuffer) {
+  memgraph::storage::PropertyStore props;
+  auto id = memgraph::storage::PropertyId::FromInt(0);
+  auto id_value = memgraph::storage::PropertyValue(1);
+  auto completion_percentage = memgraph::storage::PropertyId::FromInt(1);
+  auto completion_percentage_value = memgraph::storage::PropertyValue(14);
+  auto gender = memgraph::storage::PropertyId::FromInt(2);
+  auto gender_value = memgraph::storage::PropertyValue("man");
+  auto age = memgraph::storage::PropertyId::FromInt(3);
+  auto age_value = memgraph::storage::PropertyValue(26);
+  ASSERT_TRUE(props.SetProperty(id, id_value));
+  ASSERT_TRUE(props.SetProperty(age, age_value));
+  ASSERT_TRUE(props.SetProperty(completion_percentage, completion_percentage_value));
+  ASSERT_TRUE(props.SetProperty(gender, gender_value));
+  std::string serialized_props = memgraph::utils::SerializeProperties(props);
+  memgraph::storage::PropertyStore deserialized_props =
+      memgraph::storage::PropertyStore::CreateFromBuffer(serialized_props);
 
-// TEST_F(RocksDBStorageTest, DeserializeEdge) {
-// NOTE: This test would fail in the case of snaphsot isolation because of the way in which RocksDB
-// serializes commit timestamp.
-// auto acc = storage->Access(memgraph::storage::IsolationLevel::READ_UNCOMMITTED);
-// auto vertex1 = acc->CreateVertex();
-// auto vertex2 = acc->CreateVertex();
-// auto serialized_edge = fmt::format("{}|{}|0|1|2", vertex1.Gid().AsInt(), vertex2.Gid().AsInt());
-// auto acc_ptr = static_cast<memgraph::storage::DiskStorage::DiskAccessor *>(acc.get());
-// auto edge = acc_ptr->DeserializeEdge(serialized_edge, "garbage");
-// ASSERT_EQ(edge->Gid().AsInt(), 2);
-// ASSERT_EQ(edge->EdgeType().AsInt(), 1);
-// ASSERT_EQ(edge->from_vertex_->gid.AsInt(), vertex1.Gid().AsInt());
-// ASSERT_EQ(edge->to_vertex_->gid.AsInt(), vertex2.Gid().AsInt());
-// }
+  for (const auto &[prop_id, prop_value] : props.Properties()) {
+    ASSERT_TRUE(deserialized_props.IsPropertyEqual(prop_id, prop_value));
+  }
+}
+
+TEST_F(RocksDBStorageTest, SerializePropertiesExternalBuffer) {
+  memgraph::storage::PropertyStore props;
+  auto id = memgraph::storage::PropertyId::FromInt(0);
+  auto id_value = memgraph::storage::PropertyValue(1);
+  auto completion_percentage = memgraph::storage::PropertyId::FromInt(1);
+  auto completion_percentage_value = memgraph::storage::PropertyValue(14);
+  auto gender = memgraph::storage::PropertyId::FromInt(2);
+  auto gender_value = memgraph::storage::PropertyValue("man167863816386826");
```

Comment:
Maybe we can add a comment or empty lines to emphasize that because of this line we start using external buffer, what do you think? Because on the first glance it was hard to see the difference with previous test.

Reply:
Yes sure